docs/aws/audit/redshiftmonitoring/rules/redshift_cluster_encrypted.mdx

```diff
@@ -23,6 +23,219 @@ HIPAA, GDPR, NIST, SOC2, HITRUST, PCIDSS, NISTCSF
+   When creating a new Redshift cluster, specify the KMS key to enable encryption.
+   ```sh
+   aws redshift create-cluster \
+       --cluster-identifier my-redshift-cluster \
+       --node-type dc2.large \
+       --master-username myuser \
+       --master-user-password mypassword \
+       --cluster-type single-node \
+       --kms-key-id <KMS_KEY_ARN>
+   ```
+
+3. **Modify an Existing Redshift Cluster to Enable Encryption**:
+   If you have an existing Redshift cluster, you need to take a snapshot, then restore the cluster from the snapshot with encryption enabled.
+   ```sh
+   aws redshift create-cluster-snapshot \
+       --cluster-identifier my-redshift-cluster \
+       --snapshot-identifier my-snapshot
+   ```
+
+   ```sh
+   aws redshift restore-from-cluster-snapshot \
+       --snapshot-identifier my-snapshot \
+       --cluster-identifier my-new-redshift-cluster \
+       --kms-key-id <KMS_KEY_ARN>
+   ```
+
+4. **Verify Encryption**:
+   Finally, verify that the cluster is encrypted by describing the cluster and checking the `Encrypted` field.
+   ```sh
+   aws redshift describe-clusters \
+       --cluster-identifier my-new-redshift-cluster \
+       --query 'Clusters[*].{ClusterIdentifier:ClusterIdentifier,Encrypted:Encrypted}'
+   ```
+
+By following these steps, you can ensure that your Amazon Redshift clusters are encrypted, thereby preventing the misconfiguration of having unencrypted clusters.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent Redshift Clusters from being unencrypted in AWS Redshift using Python scripts, you can use the AWS SDK for Python (Boto3). Here are the steps to ensure that your Redshift clusters are encrypted:
+
+1. **Install Boto3**:
+   Ensure you have Boto3 installed in your Python environment. If not, you can install it using pip.
+   ```bash
+   pip install boto3
+   ```
+
+2. **Create a Boto3 Session**:
+   Set up a Boto3 session with the necessary AWS credentials and region.
+   ```python
+   import boto3
+
+   session = boto3.Session(
+       aws_access_key_id='YOUR_ACCESS_KEY',
+       aws_secret_access_key='YOUR_SECRET_KEY',
+       region_name='YOUR_REGION'
+   )
+   redshift = session.client('redshift')
+   ```
+
+3. **Check Existing Clusters for Encryption**:
+   Write a function to check if existing Redshift clusters are encrypted. If not, you can log a warning or take necessary actions.
+   ```python
+   def check_redshift_encryption():
+       clusters = redshift.describe_clusters()
+       for cluster in clusters['Clusters']:
+           if not cluster['Encrypted']:
+               print(f"Cluster {cluster['ClusterIdentifier']} is not encrypted.")
+           else:
+               print(f"Cluster {cluster['ClusterIdentifier']} is encrypted.")
+
+   check_redshift_encryption()
+   ```
+
+4. **Create Encrypted Redshift Cluster**:
+   When creating a new Redshift cluster, ensure that the `Encrypted` parameter is set to `True`.
+   ```python
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>